features/helper_specs/helper_spec.feature

```diff
@@ -120,3 +120,96 @@ Feature: helper spec
       """
     When I run `rspec spec/helpers/widgets_helper_spec.rb`
     Then the examples should all pass
+
+  Scenario: helper methods are accessible from within example
+    Given a file named "spec/helpers/user_helper_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe UserHelper, :type => :helper do
+        describe "#email" do
+          it "gives current user email" do
+            expect(email).to eq "jon.snow@example.com"
+          end
+        end
+      end
+      """
+    And a file named "app/helpers/user_helper.rb" with:
+      """ruby
+      module UserHelper
+        def email
+          "jon.snow@example.com"
+        end
+      end
+      """
+    When I run `rspec spec/helpers/user_helper_spec.rb`
+    Then the examples should all pass
+
+  Scenario: methods defined in example are visible within example
+    Given a file named "spec/helpers/user_helper_spec.rb" with:
+      """ruby
+      require "rails_helper"
+
+      RSpec.describe UserHelper, :type => :helper do
+        describe "#email" do
+          context "when user is logged in" do
+            let(:current_user) { double("user", {:email => "jon.snow@example.com"}) }
+
+            it "gives current user email" do
+              expect(email).to eq "jon.snow@example.com"
+            end
+          end
+
+          context "when user isn't logged in" do
+            def current_user; end
+
+            it "gives n/a" do
+              expect(email).to eq "n/a"
+            end
+          end
+
+          context "when current_user isn't defined" do
+            it "throws an exception" do
+              expect { email }.to raise_error NameError
+            end
+          end
+        end
+      end
+      """
+    And a file named "app/helpers/user_helper.rb" with:
+      """ruby
+      module UserHelper
+        def email
+          current_user.nil? ? 'n/a' : current_user.email
+        end
+      end
+      """
+    When I run `rspec spec/helpers/user_helper_spec.rb`
+    Then the examples should all pass
```

Comment:
This is also some very odd behavior. Combining this with the prior example, it really feels like there's a lot of state leak in these spec types. Stated a slightly different way, the barrier between test and object undercuts doesn't exist. This is bad. :frowning: 

This can easily lead to some very surprising and unexpected behaviors:

```ruby
require 'rails_helper'

module UserHelper
  def email
    :user_helper
  end
end

RSpec.describe UserHelper, :type => :helper do

  it "directly accesses the helper message" do
    expect(email).to eq :user_helper
  end

  it "the helper uses it's definition" do
    expect(helper.email).to eq :user_helper
  end

  context "with an inner context defining a conflicting let" do
    let(:email) { :let_memo }

    it "uses the let object" do
      expect(email).to eq :let_memo
    end

    it "the helper uses it's definition" do
      expect(helper.email).to eq :user_helper
    end

  end

end

RSpec.describe UserHelper, "with a conflicting let", :type => :helper do

  let(:email) { :let_memo }

  it "accesses the helper message" do
    expect(email).to eq :user_helper
  end

  it "the helper uses it's definition" do
    expect(helper.email).to eq :user_helper
  end

end

RSpec.describe UserHelper, "with a conflicting method", :type => :helper do

  def email
    :custom_method
  end

  it "accesses the conflicting message" do
    expect(email).to eq :custom_method
  end

  it "the helper uses it's definition" do
    expect(helper.email).to eq :user_helper
  end

end

RSpec.describe UserHelper, "conflicts are order dependent", :type => :helper do

  def email
    :custom_method
  end

  let(:email) { :let_memo }

  it "accesses the helper message" do
    expect(email).to eq :user_helper
  end

  it "the helper uses it's definition" do
    expect(helper.email).to eq :user_helper
  end

end

RSpec.describe UserHelper, "conflicts are order dependent", :type => :helper do

  let(:email) { :let_memo }

  def email
    :custom_method
  end

  it "accesses the conflicting message" do
    expect(email).to eq :custom_method
  end

  it "the helper uses it's definition" do
    expect(helper.email).to eq :user_helper
  end

end
```